text_generation/causal_lm/cpp/continuous_batching/library/src/sampler.hpp

```diff
@@ -295,42 +295,51 @@ class Sampler {
+            size_t sequence_offset = (seq_len - 1) * vocab_size;
+            const float* logits_data = logits.data<const float>() + batch_offset + sequence_offset;
+
+            // const float * logits_data = &logits_data_group[i * seq_tensor_size] + seq_tensor_size;
+            std::vector<LogitWithIdx> logit_vector(vocab_size);
+            for (size_t i = 0; i < logit_vector.size(); i++) {
+                logit_vector[i] = LogitWithIdx(logits_data[i], i);
+            }
 
-        auto temperature_transform = TemperatureLogitTransform(temperature);
-        std::vector<ProbabilityWithIdx> softmax_vector = temperature_transform.apply(logit_vector);
+            auto temperature_transform = TemperatureLogitTransform(temperature);
+            std::vector<ProbabilityWithIdx> softmax_vector = temperature_transform.apply(logit_vector);
 
-        std::vector<ProbabilityWithIdx> filtered(softmax_vector);
+            std::vector<ProbabilityWithIdx> filtered(softmax_vector);
 
-        if (top_p != 0.0f) {
-            auto filter = TopPFilter(top_p);
-            filtered = filter.filter(filtered);
-        }
+            if (top_p != 0.0f) {
+                auto filter = TopPFilter(top_p);
+                filtered = filter.filter(filtered);
+            }
 
-        if (top_k != 0) {
-            auto filter = TopKFilter(top_k);
-            filtered = filter.filter(filtered);
-        }
+            if (top_k != 0) {
+                auto filter = TopKFilter(top_k);
+                filtered = filter.filter(filtered);
+            }
 
-        auto normalize_transform = ProbabilityNormalizeTransform();
-        filtered = normalize_transform.apply(filtered);
-        std::vector<float> multinomial_weights(filtered.size());
-        for (size_t i = 0; i < filtered.size(); i++) multinomial_weights[i] = filtered[i].first;
+            auto normalize_transform = ProbabilityNormalizeTransform();
+            filtered = normalize_transform.apply(filtered);
+            std::vector<float> multinomial_weights(filtered.size());
+            for (size_t i = 0; i < filtered.size(); i++) multinomial_weights[i] = filtered[i].first;
 
-        auto dist = std::discrete_distribution<size_t>(multinomial_weights.begin(), multinomial_weights.end()); // equivalent to multinomial with number of trials == 1
-        size_t element_to_pick = dist(rng_engine);
-        int64_t out_token = filtered[element_to_pick].second;
+            auto dist = std::discrete_distribution<size_t>(multinomial_weights.begin(), multinomial_weights.end()); // equivalent to multinomial with number of trials == 1
+            for (size_t i = 0; i < n; ++i) {
+                size_t element_to_pick = dist(rng_engine);
+                int64_t out_token = filtered[element_to_pick].second;
+                out_tokens.push_back(out_token);
+            }
+        }
 
-        return out_token;
+        OPENVINO_ASSERT(out_tokens.size() == n || out_tokens.size() == batch_size);
```

Comment:
```suggestion
        OPENVINO_ASSERT(out_tokens.size() == n * batch_size);
```
can we move this assert to the beginning? to perform initial validation of parameters before starting actual processing